x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java

```diff
@@ -680,4 +673,42 @@ public boolean hasReferences() {
             return refs.hasReferences();
         }
     }
+
+    /**
+     * Create a {@link LookupShardContext} for a locally allocated {@link ShardId}.
+     */
+    public interface CreateShardContext {
```

Comment:
NIT: Should this be called `LookupShardContextFactory` to avoid confusion with a method name?

Reply:
Yeah, that probably would have been more clear. I can rename this in the followup PR you just reviewed.

Reply:
Thank you!